### PR TITLE
Fixes #12578: API : reload endpoints

### PR DESCRIPTION
**Pull request by Haypierre**
https://www.rudder-project.org/redmine/issues/12578

### CODE REVIEW

**Comment by Haypierre** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
You need to rebase your branch to keep only one commit

**Comment by Haypierre** [comment]
Commit modified

**Comment by Haypierre** [comment]
Commit modified

File: rudder-rest/src/main/scala/com/normation/rudder/rest/lift/SystemApi.scala [hunk 1]
```diff
@@ -27,4 +38,99 @@ class SystemApi(restExtractorService: RestExtractorService) extends LiftApiModul
...
+    updatePTLibService      : UpdateTechniqueLibrary
+  , uuidGen                 : StringUuidGenerator
+  , updateDynamicGroups     : UpdateDynamicGroups
+  , implicit val userService: UserService
+) extends Loggable {
+
+  private[this] def reloadTechniquesWrapper(req: Req) : Either[String, JField] = {
+
+    updatePTLibService.update(ModificationId(uuidGen.newUuid), getActor(req), Some("Technique library reloaded from REST API")) match {
+      case Full(x) => Right(JField("techniques", "Reloaded"))
```
Comment (VinceMacBuche) [inline rudder-rest/src/main/scala/com/normation/rudder/rest/lift/SystemApi.scala]: I think the answer should be the same in both enpoints, Put reloaded for now in both

Comment (VinceMacBuche) [reply rudder-rest/src/main/scala/com/normation/rudder/rest/lift/SystemApi.scala]: We should discuss it elsewhere on slack

File: rudder-rest/src/main/scala/com/normation/rudder/rest/lift/SystemApi.scala [hunk 2]
```diff
@@ -27,4 +38,99 @@ class SystemApi(restExtractorService: RestExtractorService) extends LiftApiModul
...
+  def reloadAll(req: Req, params: DefaultParams) = {
+
+    implicit val action = "reloadAll"
+    implicit val prettify = params.prettify
+
+    (for {
+      groups <- reloadDyngroupsWrapper()
+      techniques <- reloadTechniquesWrapper(req)
+    } yield {
+      (groups, techniques)
```
Comment (VinceMacBuche) [inline rudder-rest/src/main/scala/com/normation/rudder/rest/lift/SystemApi.scala]: You should make a List already here

**Comment by Haypierre** [comment]
Commit modified

**Comment by Haypierre** [comment]
Commit modified

**Comment by Haypierre** [comment]
Commit modified

**Pull request merged**